.github/workflows/main.yml

```diff
@@ -43,9 +39,17 @@ jobs:
         with:
           fetch-depth: 0  # Needed by diff-cover to get the changed lines: origin/master..HEAD
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+        # uses: actions/setup-python@v4 due to https://github.com/actions/setup-python/issues/672:
+        # https://github.com/MatteoH2O1999/setup-python
+        # This action tries to build from source all Python versions that actions/setup-python
+        # does not support. It also allows to cache built versions so that after the first run,
+        # installation time is really low. Hope it works also with 20.04, else it needs more changes:
```

Comment:
Have we tested this with 20.04?

Reply:
It stopped working with GitHub's new ubuntu-20.04 rootfs.

Therefore, I moved it and the other job to ubuntu-22.04.

This means all of GitHub CI is using one distro (ubuntu-22.04) as base rootfs,
which is great! Hopefully less unpleasant surprises from GitHub on that!